CMakeLists.txt

```diff
@@ -0,0 +1,109 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(backtrace)
+
+# Automake uses -frandom-seed initialized with file name of given file
+# but AFAIK it can't be done on CMake, so here's always same seed
+set(CMAKE_CXX_FLAGS "-DHAVE_CONFIG_H -funwind-tables -frandom-seed=mySeed -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual -g -O2")
+set(CMAKE_C_FLAGS "-DHAVE_CONFIG_H -funwind-tables -frandom-seed=mySeed -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual -g -O2")

```

Comment:
Public, private and interface flags should be differentiated and directly associated with the library target. For example:

```
target_compile_options(${PROJECT_NAME} PRIVATE -funwind-tables -frandom-seed=mySeed -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual)
```

* If there are compile option that must be used by project using the library, they should be declared as `INTERFACE.`  `INTERFACE` means that project linking against the `libbacktrace` [imported target](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#imported-targets) will transitively  use the flag.

* `HAVE_CONFIG_H` is not used in the project and should probably be removed.

* `-g -O2` flags are automatically listed when building the project with `RELWITHDEBINFO` build type. Is this a strong requirement ? 

The default flags for each build type are available here:

https://github.com/Kitware/CMake/blob/da3dc2f0cfb8e2aed207c21e419a60525eea0c6f/Modules/Compiler/GNU.cmake#L42-L47